B/bliss/build_tarballs.jl

```diff
@@ -3,30 +3,29 @@
 using BinaryBuilder, Pkg
 
 name = "bliss"
-version = v"0.73.1" # <-- This is a lie, we're bumping from 0.73 to 0.73.1 to create a Julia v1.6+ release with experimental platforms
+version = v"0.77.0"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("http://www.tcs.hut.fi/Software/bliss/bliss-0.73.zip", "f57bf32804140cad58b1240b804e0dbd68f7e6bf67eba8e0c0fa3a62fd7f0f84"),
+    ArchiveSource("https://users.aalto.fi/~tjunttil/bliss/downloads/bliss-0.77.zip", "acc8b98034f30fad24c897f365abd866c13d9f1bb207e398d0caf136875972a4"),
     DirectorySource("./bundled"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir
-cd bliss-0.73
-if [[ "${target}" == *mingw* ]]; then
-  atomic_patch -p1 ../patches/notimer.patch
-fi
-# build with GMP and store this information
-sed -i -e 's/^namespace bliss/#define BLISS_USE_GMP\n\nnamespace bliss/' defs.hh
```

Comment:
Don't forget to install the headers as well and please keep this line or at least something adapted from it. Otherwise it will be very difficult for any downstream packages to know whether bliss was built with or without GMP (and the wrong choice will produce segfaults at runtime), this just inserts that definition into one main header file and will make sure anything depending on bliss also uses that flag during compilation.

Reply:
thanks for the headers. For GMP, it seems the new version already adds this flag to defs.hh
```
/workspace/srcdir/bliss-0.77/src/defs.hh:33:9: warning: 'BLISS_USE_GMP' macro redefined [-Wmacro-redefined]
#define BLISS_USE_GMP
        ^
<command line>:3:9: note: previous definition is here
#define BLISS_USE_GMP 1
        ^
1 warning generated.
```

Reply:
```
/workspace/srcdir/bliss-0.77/src/defs.hh:33:9: warning: 'BLISS_USE_GMP' macro redefined [-Wmacro-redefined]
#define BLISS_USE_GMP
        ^
<command line>:3:9: note: previous definition is here
#define BLISS_USE_GMP 1
        ^
1 warning generated.
```
This warning is a conflict between a flag from the command line and the one added to defs.hh. Setting `-DUSE_GMP=on` for cmake seems to add `-DBLISS_USE_GMP=1` to the CFLAGS, but this is not stored anywhere for later usage. To avoid the warning you could patch this after building the library.

Reply:
couldn't we just use `#ifdef BLISS_USE_GMP`?

Reply:
Something like the following would also work yes (and even avoid warnings if anyone downstream manually sets BLISS_USE_GMP).
```c
#ifndef BLISS_USE_GMP
#define BLISS_USE_GMP 1
#endif
```

(Most other software packages generate some config.h from a config.h.in which contains such options, or generate a pkg-config files which contains the flag in the CFLAGS)